CrossPlatformDesktopProject/CrossPlatformDesktopProject/Levels/Background.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.Levels
+{
+    class Background
+    {
+        private int screenWidth;
+        private int screenHeight;
+        private int innerWidth;
+        private int innerHeight;
+        private int destinationWidth;
+        private int destinationHeight;
+        private int doorWidth;
+        private int doorHeight;
+        private string roomTextureID;
+        private Rectangle topDoorDest;
+        private Rectangle leftDoorDest;
+        private Rectangle rightDoorDest;
+        private Rectangle bottomDoorDest;
+
+        public Background(Game1 game, string texture) 
+        {
+            screenWidth = game.GraphicsDevice.Viewport.Width;
+            screenHeight = game.GraphicsDevice.Viewport.Height;
+            innerWidth = (screenWidth * 191) / 255 + 6;
+            innerHeight = (screenHeight * 111) / 175 + 6;
+            destinationHeight = (screenHeight - innerHeight) / 2 + 1;
+            destinationWidth = (screenWidth - innerWidth) / 2 + 1;
+            roomTextureID = texture;
+
+            doorWidth = (screenWidth * 31) / 255;
+            doorHeight = (screenHeight * 31) / 175;
+            //screenWidth - (doorWidth / 2)= 7
+            topDoorDest = new Rectangle((screenWidth/2) - (doorWidth / 2), 0, doorWidth, doorHeight);
+            leftDoorDest = new Rectangle(0, (screenHeight / 2) - (doorHeight / 2) + 1, doorWidth, doorHeight);
+            rightDoorDest = new Rectangle(screenWidth - doorWidth, (screenHeight / 2) - (doorHeight / 2), doorWidth, doorHeight);
+            bottomDoorDest = new Rectangle((screenWidth / 2) - (doorWidth / 2), screenHeight - doorHeight, doorWidth, doorHeight);
+        }
+
+        public void Draw(SpriteBatch sb, string[] adjacents)
+        {
+            RoomTextureStorage textureInstance = RoomTextureStorage.Instance;
+            Texture2D texture = textureInstance.getRoomSpriteSheet();
+
+            Rectangle borderSource = RoomTextureStorage.BORDER;
+            Rectangle borderDestination = new Rectangle(0, 0, screenWidth, screenHeight);
+            sb.Draw(texture, borderDestination, borderSource, Color.White);
+            //Recover source image from CSV string
+
+            DoorCheck(sb, adjacents, texture);
+
+            FieldInfo propInfo = typeof(RoomTextureStorage).GetField(roomTextureID);
+            object rectType = propInfo.GetValue(typeof(RoomTextureStorage));
+            Rectangle source = (Rectangle)rectType;
+
+            Rectangle destination = new Rectangle(destinationWidth, destinationHeight, innerWidth, innerHeight);
+            sb.Draw(texture, destination, source, Color.White);
+        }
+
+        public void DoorCheck(SpriteBatch sb, string[] adjacentRooms, Texture2D texture)
+        {
+            if (adjacentRooms[0].Equals("-1"))
```

Comment:
This series of if/elses works, but could probably be cleaned up to make it less redundant and a bit cleaner. Not urgent.

Reply:
Definitely agree here. This is the worst portion of code in the branch. I just struggled with standardizing the check for all 4 doors, because they all need to access the pre-recorded Rectangle property of RoomTextureStorage with different names.

Reply:
Which seems like it would be a hefty task for Reflection and possibly impact performance(?) 
- I don't understand reflection as well as I should, so I'm not sure.